code/game/machinery/big_manipulator/big_manipulator.dm

```diff
@@ -0,0 +1,909 @@
+			return TRUE
+
+		if("add_atom_filter_from_held")
+			var/obj/item/held_item = user.get_active_held_item()
+
+			if(!held_item)
+				return FALSE
+
+			for(var/filter_path in target_point.atom_filters)
+				if(istype(held_item, filter_path))
+					return FALSE
+
+			target_point.atom_filters += held_item.type
+			return TRUE
+
+		if("delete_filter")
+			target_point.atom_filters.Cut(value, value + 1)
+			return TRUE
+
+		if("toggle_worker_rmb")
+			target_point.worker_use_rmb = !target_point.worker_use_rmb
+			return TRUE
+
+		if("toggle_worker_combat")
+			target_point.worker_combat_mode = !target_point.worker_combat_mode
+			return TRUE
+
+		if("priority_move_up")
+			return target_point.move_priority_up_by_index(value)
+
+		if("move_to")
+			var/button_number = text2num(value["buttonNumber"])
+
+			var/dx = 0
+			var/dy = 0
+			switch(button_number)
+				if(1)
+					dx = -1
+					dy = 1
+				if(2)
+					dx = 0
+					dy = 1
+				if(3)
+					dx = 1
+					dy = 1
+				if(4)
+					dx = -1
+					dy = 0
+				if(5)
+					dx = 0
+					dy = 0
+				if(6)
+					dx = 1
+					dy = 0
+				if(7)
+					dx = -1
+					dy = -1
+				if(8)
+					dx = 0
+					dy = -1
+				if(9)
+					dx = 1
+					dy = -1
+
+			var/turf/new_turf = locate(x + dx, y + dy, z)
+			if(!new_turf || isclosedturf(new_turf))
+				return FALSE
+
+			target_point.interaction_turf = new_turf
+			update_hud()
+			return TRUE
+
+/// Cycles the given value in the given list. Retuns the next value in the list, or the first one if the list isn't long enough.
+/obj/machinery/big_manipulator/proc/cycle_value(current_value, list/possible_values)
+	var/current_index = possible_values.Find(current_value)
+	if(current_index == 0)
+		return possible_values[1]
+
+	var/next_index = (current_index % length(possible_values)) + 1
+	return possible_values[next_index]
+
+/// Begins a new task with the specified type and duration
+/obj/machinery/big_manipulator/proc/start_task(task_type, duration)
+	if(current_task == CURRENT_TASK_STOPPING)
+		return
+
+	end_current_task() // ends any previous task first (momentarily sets IDLE)
+	current_task_start_time = world.time
+	current_task_duration = duration / 10 // the duration is in deciseconds for TGUI
+	current_task = task_type
+	SStgui.update_uis(src)
+
+/// Ends the current task
+/obj/machinery/big_manipulator/proc/end_current_task()
+	current_task_start_time = 0
+	current_task_duration = 0
+	if(current_task == CURRENT_TASK_STOPPING)
+		current_task = CURRENT_TASK_NONE
+	SStgui.update_uis(src) // Update UI immediately
+
+/// Completes the stopping task and transitions to TASK_NONE
+/obj/machinery/big_manipulator/proc/complete_stopping_task()
+	if(current_task == CURRENT_TASK_STOPPING)
+		on = FALSE
+		cycle_timer_running = FALSE
+		end_current_task()
+		SStgui.update_uis(src)
+
+/obj/machinery/big_manipulator/proc/remove_all_huds()
+	for(var/image/hud_image in hud_points)
+		qdel(hud_image)
```

Comment:
please don't qdel images it doesn't do anything but hard delete them, need to actually clean up the hud "construction" or whatever